src/test/java/com/rocketden/main/socket/GameSocketTests.java

```diff
@@ -0,0 +1,145 @@
+
+    @BeforeEach
+    public void setup() throws Exception {
+        baseRestEndpoint = "http://localhost:" + port + "/api/v1/rooms";
+
+        UserDto host = new UserDto();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+        CreateRoomRequest createRequest = new CreateRoomRequest();
+        createRequest.setHost(host);
+
+        // Create room
+        HttpEntity<CreateRoomRequest> createEntity = new HttpEntity<>(createRequest);
+        room = template.postForObject(baseRestEndpoint, createEntity, RoomDto.class);
+        assertNotNull(room);
+
+        UserDto user = new UserDto();
+        user.setNickname(NICKNAME_2);
+        user.setUserId(USER_ID_2);
+        JoinRoomRequest joinRequest = new JoinRoomRequest();
+        joinRequest.setUser(user);
+
+        // Join room
+        HttpEntity<JoinRoomRequest> joinEntity = new HttpEntity<>(joinRequest);
+        String joinRoomEndpoint = String.format("%s/%s/users", baseRestEndpoint, room.getRoomId());
+        room = template.exchange(joinRoomEndpoint, HttpMethod.PUT, joinEntity, RoomDto.class).getBody();
+        assertNotNull(room);
+
+        // Update room settings
+        UpdateSettingsRequest updateRequest = new UpdateSettingsRequest();
+        updateRequest.setInitiator(host);
+        updateRequest.setDuration(DURATION);
+        HttpEntity<UpdateSettingsRequest> updateEntity = new HttpEntity<>(updateRequest);
+        String updateEndpoint = String.format("%s/%s/settings", baseRestEndpoint, room.getRoomId());
+        room = template.exchange(updateEndpoint, HttpMethod.PUT, updateEntity, RoomDto.class).getBody();
+        assertNotNull(room);
+
+        // Create problems
+        SocketTestMethods.createSingleProblemAndTestCases(template, port);
+
+        // Start game
+        StartGameRequest startRequest = new StartGameRequest();
+        startRequest.setInitiator(host);
+        HttpEntity<StartGameRequest> startEntity = new HttpEntity<>(startRequest);
+        String startEndpoint = String.format("%s/%s/start", baseRestEndpoint, room.getRoomId());
+        room = template.exchange(startEndpoint, HttpMethod.POST, startEntity, RoomDto.class).getBody();
+        assertNotNull(room);
+
+        // Set up the socket connection and subscription
+        blockingQueue = new ArrayBlockingQueue<>(2);
+        hostSession = SocketTestMethods.connectToSocket(CONNECT_ENDPOINT, USER_ID, this.port);
+
+        hostSession.subscribe(String.format(SUBSCRIBE_ENDPOINT, room.getRoomId()), new StompFrameHandler() {
+            @Override
+            public Type getPayloadType(StompHeaders headers) {
+                return GameDto.class;
+            }
+
+            @Override
+            public void handleFrame(StompHeaders headers, Object payload) {
+                blockingQueue.add((GameDto) payload);
+            }
+        });
+    }
+
+    @Test
+    public void socketReceivesMessageOnGameOver() throws Exception {
```

Comment:
Thanks for making this! I should've done it in a previous PR, sorry about that -- appreciate it! 👍